garage/tf/models/base.py

```diff
@@ -0,0 +1,84 @@
+"""This file contains the abstraction class for models."""
+from garage.tf.models import AutoPickable
+
+
+class BaseModel:
+    """Abstraction class for models."""
+
+    def build_model(self):
+        """
+        Build model.
+
+        This function will build the whole graph for
+        the model. All tensors should be created here.
+        By calling this function, a copy of the same graph
+        should be created with the same parameters.
+        """
+        raise NotImplementedError
+
+    @property
+    def input(self):
+        """Tensor input of the Model."""
+        if len(self.model.inputs) == 1:
+            return self.model.inputs[0]
+
+        return self.model.inputs
+
+    @property
+    def output(self):
+        """Tensor output of the Model."""
+        if len(self.model.outputs) == 1:
+            return self.model.outputs[0]
+
+        return self.model.outputs
+
+    @property
+    def inputs(self):
```

Comment:
Any reason not merge input and inputs into a single interface?

Reply:
comply with `tf.keras.model`

Reply:
If we just want to return the config for the underlying model, since we construct every model with the Functional API I think we can do the "Aggregation" way:
```python
class SomeModel:
    def __init__(...):
        self.model = self.build_model()

    @property
    def input:
        return self.model.input

    def get_config():
        return self.model.get_config()
```
But if we want to return the config that reflects the entire model, I think that's not easy.

Reply:
i c. so the pickler of `GaussianMLPModel` actually pickles the config and weights of `GaussianMLPModel.model` instead of `GaussianMLPModel`?

Reply:
Yes. `GaussianMLPModel` is the abstract class that holds the underlying `GaussianMLPModel` as `self.model`.

Reply:
Wait, does `tf.keras.Model` have both `input` and `inputs` properties? If so, can we make it unnecessary to implement both (probably `input`)?

Reply:
Yes, it does.